src/redact/index.ts

```diff
@@ -21,3 +24,31 @@ export const addDefaultRedactPathStrings = (
   }
   return redact;
 };
+
+export const addRemovePathsCensor = (
+  redact: ExtendedRedact,
+): ExtendedRedact => {
+  if (
+    !redact ||
+    Array.isArray(redact) ||
+    !redact.removePaths ||
+    redact.removePaths.length === 0
+  ) {
+    return redact;
+  }
+
+  const { paths: redactPaths, removePaths } = redact;
+
+  const redactMap = redactPaths.reduce(
+    (previous, current) => ({ ...previous, [current]: true }),
+    {} as Record<string, boolean>,
+  );
```

Comment:
A `Set` seems like the more appropriate data structure

Reply:
Does it perform as well as an object for reads?

Reply:
`Set` is a much narrower purpose data structure so I'd assume it'd be faster. Looks like it's benchmarked to be about 2x as fast with half the memory usage in Node:

<https://github.com/anvaka/set-vs-object>

Reply:
Thanks, I've changed to a `Set`.